src/Core/Form/IdentifiableObject/DataProvider/CartRuleFormDataProvider.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataProvider;
+
+use DateTimeImmutable;
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Domain\CartRule\Query\GetCartRuleForEditing;
+use PrestaShop\PrestaShop\Core\Domain\CartRule\QueryResult\EditableCartRule;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Reduction;
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime as DateTimeUtil;
+
+class CartRuleFormDataProvider implements FormDataProviderInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    public function __construct(
+        CommandBusInterface $queryBus,
+        ConfigurationInterface $configuration
+    ) {
+        $this->queryBus = $queryBus;
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getData($id)
+    {
+        /** @var EditableCartRule $editableCartRule */
+        $editableCartRule = $this->queryBus->handle(new GetCartRuleForEditing($id));
+
+        //@todo: finish up in a dedicated PR when EditCartRuleCommand is introduced
+        return [
+            'information' => [
+                'name' => $editableCartRule->getInformation()->getLocalizedNames(),
+            ],
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getDefaultData()
+    {
+        //@todo: we should consider introducing some clockInterface so we can mock "now" in tests
+        $now = new DateTimeImmutable();
```

Comment:
If you want to create a unit test for this, you can create a protected method `getNow()` that will return `new DateTimeImmutable`

Once you test this method, you can create a test class, and replace that method
```suggestion
        $now = $this->getNow();
```

```php
class CartRuleFormDataProvider
{
	protected function getNow()
	{
		return new DateTimeImmutable();
	}
}

class CartRuleFormDataProviderTest extends CartRuleFormDataProvider 
{
	protected function getNow()
	{
		return new DateTimeImmutable('2020-01-01');
	}
}
```

Reply:
thats an option, but it is just a workaround :(